### PR TITLE
[BEAM-13293] XLang Jdbc IO for Go SDK

### PR DESCRIPTION
**Pull request by riteshghorse**
Xlang jdbc io for Go SDK. The current version has support for limited types in schema for Read Transform.
See [BEAM-13717](https://issues.apache.org/jira/browse/BEAM-13717) for more details.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

`ValidatesRunner` compliance status (on master branch)
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_ULR/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Examples testing status on various runners
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Java11_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Post-Commit SDK/Transform Integration Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Go</th>
      <th>Java</th>
      <th>Python</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
  </tbody>
</table>

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>---</th>
      <th>Java</th>
      <th>Python</th>
      <th>Go</th>
      <th>Website</th>
      <th>Whitespace</th>
      <th>Typescript</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Non-portable</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Portable</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_GoPortable_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by youngoli** [comment]
I think I figured out the problem you're having. I noticed that when I ran this, the SDK harness was outputting "PASS". This means it's running tests in the SDK harness instead of acting as an SDK harness, which led me to find the culprit:

You need the following code in your test file:
```
func TestMain(m *testing.M) {
	ptest.Main(m)
}
```
To give a short version of why, the SDK harness is actually executing the exact binary that the code is originally executed from. Normally in our files we have a call in beam.Init to check if it's the SDK harness or user code, and if the binary is run as the SDK harness it behaves differently.

To get the same thing working for our tests, we need to replace TestMain with ptest.Main, otherwise it will try to execute all the tests before we can run beam.Init. ptest.Main handles that for us by calling beam.Init before deferring to the normal TestMain behavior. This means any kind of integration test we write for the Go SDK needs to call that snippet, or something equivalent, because we absolutely need beam.Init to get called before going to the normal test framework logic, otherwise SDK harnesses built for those integration tests will attempt to run tests instead of actually act as an SDK harness.

File: sdks/go/test/integration/io/xlang/jdbc/jdbc.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	jdbcio.Write(s, expansionAddr, tableName, driverClassName, jdbcUrl, username, password, rows) //, jdbcio.WriteStatement(statement))
+}
+
+// WritePipeline creates a pipeline that writes a given slice of ints to Kafka.
+func WritePipeline(expansionAddr, tableName, driverClassName, jdbcUrl, username, password string) *beam.Pipeline {
+	beam.Init()
+	p, s := beam.NewPipelineWithRoot()
+	rows := []JdbcWriteTestRow{{"row1"}, {"row2"}}
+
+	writeRows(s, expansionAddr, tableName, driverClassName, jdbcUrl, username, password, beam.Create(s, rows))
```
Comment (lostluck) [inline sdks/go/test/integration/io/xlang/jdbc/jdbc.go:47]: The following should resolve the coder issue in https://issues.apache.org/jira/browse/BEAM-13418

```suggestion
	input :=  beam.CreateList(s, rows)
	writeRows(s, expansionAddr, tableName, driverClassName, jdbcUrl, username, password, input)
```

**Comment by lostluck** [comment]
Retest this please

**Comment by lostluck** [comment]
Re-pinging the tests since everything in the repo has changed since this was last tested.

**Comment by riteshghorse** [comment]
R: @lostluck @youngoli 
CC: @jrmccluskey

**Comment by lostluck** [comment]
FYI Go Portable failed with the new test. Looking at the code now for any insights I can provide.

```
--- FAIL: TestJDBCIO_BasicReadWrite (8.57s)
panic: 	tried cross-language for beam:transform:org.apache.beam:schemaio_jdbc_write:v1 against localhost:9000 and failed
	expanding external transform
	expanding transform with ExpansionRequest: components:{pcollections:{key:"n2" value:{unique_name:"n2" coder_id:"c0@nhyUeOslCm" is_bounded:BOUNDED windowing_strategy_id:"w0@nhyUeOslCm"}} windowing_strategies:{key:"w0@nhyUeOslCm" value:{window_fn:{urn:"beam:window_fn:global_windows:v1"} merge_status:NON_MERGING window_coder_id:"c1@nhyUeOslCm" trigger:{default:{}} accumulation_mode:DISCARDING output_time:END_OF_WINDOW closing_behavior:EMIT_IF_NONEMPTY OnTimeBehavior:FIRE_IF_NONEMPTY environment_id:"go"}} coders:{key:"c0@nhyUeOslCm" value:{spec:{urn:"beam:coder:row:v1" payload:"\n\r\n\x07role_id\x1a\x02\x10\x03\x12$abc5e222-ab8f-4f6f-800a-a58bb0f45675"}}} coders:{key:"c1@nhyUeOslCm" value:{spec:{urn:"beam:coder:global_window:v1"}}} environments:{key:"go" value:{}}} transform:{unique_name:"External" spec:{urn:"beam:transform:org.apache.beam:schemaio_jdbc_write:v1" payload:"\nX\n\x0e\n\x08location\x1a\x02\x10\x07\n\x0c\n\x06config\x1a\x02\x10\t\n\x12\n\ndataSchema\x1a\x04\x08\x01\x10\t\x12$853bc92e-cc52-429e-bf30-f534bebcedce\x12`\x03\x01\x04\x05rolesV\n\x02\xf0\x03\x15org.postgresql.Driver*jdbc:postgresql://localhost:56085/postjdbc\x07newuser\x08password"} inputs:{key:"i0" value:"n2"} environment_id:"go"} namespace:"nhyUeOslCm"
expansion failed
	caused by:
rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:9000: connect: connection refused" [recovered]
	panic: 	tried cross-language for beam:transform:org.apache.beam:schemaio_jdbc_write:v1 against localhost:9000 and failed
	expanding external transform
	expanding transform with ExpansionRequest: components:{pcollections:{key:"n2" value:{unique_name:"n2" coder_id:"c0@nhyUeOslCm" is_bounded:BOUNDED windowing_strategy_id:"w0@nhyUeOslCm"}} windowing_strategies:{key:"w0@nhyUeOslCm" value:{window_fn:{urn:"beam:window_fn:global_windows:v1"} merge_status:NON_MERGING window_coder_id:"c1@nhyUeOslCm" trigger:{default:{}} accumulation_mode:DISCARDING output_time:END_OF_WINDOW closing_behavior:EMIT_IF_NONEMPTY OnTimeBehavior:FIRE_IF_NONEMPTY environment_id:"go"}} coders:{key:"c0@nhyUeOslCm" value:{spec:{urn:"beam:coder:row:v1" payload:"\n\r\n\x07role_id\x1a\x02\x10\x03\x12$abc5e222-ab8f-4f6f-800a-a58bb0f45675"}}} coders:{key:"c1@nhyUeOslCm" value:{spec:{urn:"beam:coder:global_window:v1"}}} environments:{key:"go" value:{}}} transform:{unique_name:"External" spec:{urn:"beam:transform:org.apache.beam:schemaio_jdbc_write:v1" payload:"\nX\n\x0e\n\x08location\x1a\x02\x10\x07\n\x0c\n\x06config\x1a\x02\x10\t\n\x12\n\ndataSchema\x1a\x04\x08\x01\x10\t\x12$853bc92e-cc52-429e-bf30-f534bebcedce\x12`\x03\x01\x04\x05rolesV\n\x02\xf0\x03\x15org.postgresql.Driver*jdbc:postgresql://localhost:56085/postjdbc\x07newuser\x08password"} inputs:{key:"i0" value:"n2"} environment_id:"go"} namespace:"nhyUeOslCm"
expansion failed
	caused by:
rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:9000: connect: connection refused"

goroutine 13 [running]:
testing.tRunner.func1.2(0xe57340, 0xc0000acb80)
	/home/jenkins/sdk/go1.16.12/src/testing/testing.go:1143 +0x332
testing.tRunner.func1(0xc000102c00)
	/home/jenkins/sdk/go1.16.12/src/testing/testing.go:1146 +0x4b6
panic(0xe57340, 0xc0000acb80)
	/home/jenkins/sdk/go1.16.12/src/runtime/panic.go:965 +0x1b9
github.com/apache/beam/sdks/v2/go/pkg/beam.CrossLanguage(0xc000454ca0, 0xc0000b82d0, 0xfa2085, 0x35, 0xc000182240, 0xbc, 0xbc, 0x7ffee806128c, 0xe, 0xc0006b3b38, ...)
	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_GoPortable_Commit/src/sdks/go/pkg/beam/xlang.go:162 +0x1f7
github.com/apache/beam/sdks/v2/go/pkg/beam/io/xlang/jdbcio.Write(0xc000454c80, 0xc0000b82d0, 0x7ffee806128c, 0xe, 0xf74220, 0x5, 0xf84670, 0x15, 0xc00055a840, 0x2a, ...)
	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_GoPortable_Commit/src/sdks/go/pkg/beam/io/xlang/jdbcio/jdbc.go:129 +0x4c7
github.com/apache/beam/sdks/v2/go/test/integration/io/xlang/jdbc.writeRows(0xc000454be0, 0xc0000b82d0, 0x7ffee806128c, 0xe, 0xf74220, 0x5, 0xf84670, 0x15, 0xc00055a840, 0x2a, ...)
	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_GoPortable_Commit/src/sdks/go/test/integration/io/xlang/jdbc/jdbc.go:39 +0x1be
github.com/apache/beam/sdks/v2/go/test/integration/io/xlang/jdbc.WritePipeline(0x7ffee806128c, 0xe, 0xf74220, 0x5, 0xf84670, 0x15, 0xc00055a840, 0x2a, 0xf76805, 0x7, ...)
	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_GoPortable_Commit/src/sdks/go/test/integration/io/xlang/jdbc/jdbc.go:49 +0x20d
github.com/apache/beam/sdks/v2/go/test/integration/io/xlang/jdbc.TestJDBCIO_BasicReadWrite(0xc000102c00)
	/home/jenkins/jenkins-slave/workspace/beam_PreCommit_GoPortable_Commit/src/sdks/go/test/integration/io/xlang/jdbc/jdbc_test.go:105 +0x227
testing.tRunner(0xc000102c00, 0xfb8480)
	/home/jenkins/sdk/go1.16.12/src/testing/testing.go:1193 +0xef
created by testing.(*T).Run
	/home/jenkins/sdk/go1.16.12/src/testing/testing.go:1238 +0x2b3
FAIL	github.com/apache/beam/sdks/v2/go/test/integration/io/xlang/jdbc	8.597s
```

File: sdks/go/test/run_validatesrunner_tests.sh [hunk 2]
```diff
@@ -255,6 +255,7 @@ elif [[ "$RUNNER" == "flink" || "$RUNNER" == "spark" || "$RUNNER" == "samza" ||
   if [[ -z "$ENDPOINT" ]]; then
     JOB_PORT=$(python3 -c "$SOCKET_SCRIPT")
     ENDPOINT="localhost:$JOB_PORT"
+    IO_EXPANSION_ADDR="localhost:9000"
```
Comment (lostluck) [inline sdks/go/test/run_validatesrunner_tests.sh:258]: I suspect the issue is you're hard coding a port. Use the same format as JOB_PORT to get a new one, and then use that.

File: sdks/java/container/boot.go [hunk 3]
```diff
@@ -138,6 +138,7 @@ func main() {
 		filepath.Join(jarsDir, "beam-sdks-java-harness.jar"),
 		filepath.Join(jarsDir, "beam-sdks-java-io-kafka.jar"),
 		filepath.Join(jarsDir, "kafka-clients.jar"),
+		filepath.Join(jarsDir, "postgresql.jar"),
```
Comment (lostluck) [inline sdks/java/container/boot.go:141]: If the python tests work already is this line even required? Isn't this supposed to be handled by he artifact fetching step? 
We may want to clarify this @chamikaramj and @ihji. Does adding this line here mean we're requiring all Java jobs to have this Jar?

**Review by lostluck — COMMENTED** [review]
Go looks fine, other than the test failure and port. Tagged in cross language folks more familiar with Java to comment on the boot.go and gradle changes.

File: sdks/java/container/boot.go (hunk 3 above)
Comment (riteshghorse) [reply sdks/java/container/boot.go:141]: We don't need it. I figured it out later that the corresponding JAR will be pulled up that matches the driver name used in Read/Write transform.

File: sdks/go/test/integration/io/xlang/jdbc/jdbc_test.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+func setupTestContainer(t *testing.T, dbname, username, password string) int {
+	t.Helper()
+
+	var env = map[string]string{
+		"POSTGRES_PASSWORD": password,
+		"POSTGRES_USER":     username,
+		"POSTGRES_DB":       dbname,
+	}
+
+	var port = "5432/tcp"
```
[the remark this answers is not in the document]
Comment (riteshghorse) [reply sdks/go/test/integration/io/xlang/jdbc/jdbc_test.go:62]: 5432 is the default port for postgresql. Tried changing it with open port but then the container was not starting.

**Comment by lostluck** [comment]
Run Go Flink ValidatesRunner

**Review by lostluck — APPROVED** [review]
Thanks! LGTM

**Pull request merged**